boringtun/src/ffi/mod.rs

```diff
@@ -160,6 +163,63 @@ pub unsafe extern "C" fn check_base64_encoded_x25519_key(key: *const c_char) ->
     }
 }
 
+/// Custom tracing_subscriber writer to an external function pointer
+struct FFIFunctionPointerWriter {
+    log_func: unsafe extern "C" fn(*const c_char),
+}
+
+/// Implements Write trait for use with tracing_subscriber
+impl Write for FFIFunctionPointerWriter {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, std::io::Error> {
+        let out_str = String::from_utf8_lossy(buf).to_string();
+        let c_string = CString::new(format!("{}", out_str)).unwrap();
+        unsafe { (self.log_func)(c_string.as_ptr()) }
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> Result<(), std::io::Error> {
+        // no-op
+        Ok(())
+    }
+}
+
+/// Sets the default tracing_subscriber to write to log_func.
+/// Uses Compact format without level, target, thread ids, thread names, or ansi control characters.
+/// Subscribes to TRACE level events.
+/// Returns false on failure.
```

Comment:
You should probably document some of the reasons that "failure" might occur. For starters, it can happen when you try to set the logger twice.